src/lib.rs

```diff
@@ -169,41 +173,43 @@ fn now() -> Instant {
 /// There are different ways to create the queue:
 ///
 /// ```
+/// use std::time::Duration;
 /// use sum_queue::SumQueue;
+///
 /// let mut queue: SumQueue<i32>;
 ///
 /// // Create a queue with elements that expires after 60 seconds
-/// queue = SumQueue::new(60);
-/// // Create with 60 secs expiration and an initial capacity of 20 elements
-/// queue = SumQueue::with_capacity(60, 20);
+/// queue = SumQueue::new(Duration::from_secs(60));
+/// // Create with 500 milliseconds expiration and an initial capacity of 20 elements
+/// queue = SumQueue::with_capacity(Duration::from_millis(500), 20);
 /// ```
 pub struct SumQueue<T> {
     /// the heap with the data
     heap: BinaryHeap<QueueElement<T>>,
-    /// max time in seconds the elements will
+    /// max time the elements will
     /// live in the queue.
     max_age: Duration,
 }
 
 impl<T> SumQueue<T> {
     /// Creates an empty `SumQueue`, where the elements inside
-    /// will live `max_age_secs` seconds at maximum.
-    pub fn new(max_age_secs: u64) -> SumQueue<T> {
+    /// will live `max_age_duration` at maximum.
+    pub fn new(max_age_duration: Duration) -> SumQueue<T> {
         SumQueue {
             heap: BinaryHeap::<QueueElement<T>>::new(),
-            max_age: Duration::from_secs(max_age_secs),
+            max_age: max_age_duration,
```

Comment:
minor: the "_duration" part of the variable name is not necessary since the data type tells us everything we need to know.

Reply:
Yes, the problem is that Rust does not have formal constructors where to use the `self` reference like:

```rust
pub fn new(&mut self, max_age: Duration) -> SumQueue<T> {
    self.heap = BinaryHeap::<QueueElement<T>>::new();
    self.max_age = max_age;
}
```

In the case above `max_age` is not the same than `self.max_age`. So I have to use a name other than `max_age` for the argument. Is there a better way to name the variable that you can suggest? I just took a look to the Rust doc to refresh my knowledge but couldn't find a better way to deal with this situation.